docs/overview/Lifecycle.md

```diff
@@ -0,0 +1,335 @@
+also utilizes. We can configure the ports to use for the
+development CDN and proxy servers ran alongside One App. By
+default, One App is ran on port `3000`, the [`@americanexpress/one-app-dev-cdn`](one-app-dev-cdn) development CDN
+is ran on port `3001` and the development proxy server [`@americanexpress/one-app-dev-proxy`](one-app-dev-proxy)
+is ran on port `3002`. To modify any of the three ports:
+
+>  * [`HTTP_PORT`](../api/server/environment-variables#http_port)
+>  * [`HTTP_ONE_APP_DEV_CDN_PORT`](../api/server/environment-variables#http_one_app_dev_cdn_port)
+>  * [`HTTP_ONE_APP_DEV_PROXY_SERVER_PORT`](../api/server/environment-variables#http_one_app_dev_proxy_server_port)
+
+There is a metrics server that runs in development and production,
+which tracks certain activities happening internally and reports the
+metrics on demand (during local development `http://localhost:3005/metrics`).
+The metrics server runs on port `3005` by default, to change the port:
+
+>  * [`HTTP_METRICS_PORT`](../api/server/environment-variables#http_metrics_port)
+
+**Production Environment**
+
+There are mandatory environment variables that need to be supplied to run One App
+and there are others to configure aspects of the One App runtime such as reporting
+URLs for errors and CSP violations that occurred on the client. You will need to
+include the environment variables below to run One App in production mode:
+
+>  * [`ONE_CLIENT_CDN_URL`](../api/server/environment-variables#one_client_cdn_url) ⚠️
+>  * [`ONE_CLIENT_REPORTING_URL`](../api/server/environment-variables#one_client_reporting_url) ⚠️
+>  * [`ONE_CLIENT_CSP_REPORTING_URL`](../api/server/environment-variables#one_client_csp_reporting_url) ⚠️
+
+While `HOLOCRON_MODULE_MAP_URL` is a required environment variable in `production`,
+we will go into greater depths on this variable [in the next section on Holocron runtime.](#holocron-runtime)
+
+>  * [`HOLOCRON_MODULE_MAP_URL`](../api/server/environment-variables#holocron_module_map_url) ⚠️
+
+As we get to production, we can configure a certificate for One App
+to use and the HTTPS `port` provided (no defaults) for One App. In addition,
+the IP address can be set as well (defaults to `0.0.0.0` during production):
+
+>  * [`IP_ADDRESS`](../api/server/environment-variables#ip_address)
+>  * [`HTTPS_PORT`](../api/server/environment-variables#https_port) ⚠️
+>  * [`HTTPS_PRIVATE_KEY_PASS_FILE_PATH`](../api/server/environment-variables#https_private_key_pass_file_path) ⚠️
+>  * [`HTTPS_PRIVATE_KEY_PATH`](../api/server/environment-variables#https_private_key_path) ⚠️
+>  * [`HTTPS_PUBLIC_CERT_CHAIN_PATH`](../api/server/environment-variables#https_public_cert_chain_path) ⚠️
+>  * [`HTTPS_TRUSTED_CA_PATH`](../api/server/environment-variables#https_trusted_ca_path) ⚠️
+
+### Holocron Runtime
+
+Before One App can start rendering HTML documents from the server, the app will
+preload every Holocron module defined in the Holocron module map. This module map
```

Comment:
can we also add that the modules locally can be defined under package.json?

Reply:
I think we cover that in the "Getting Started" guide, which would come before this guide (under [`Running One App - Local Configuration`](https://github.com/americanexpress/one-app/blob/main/docs/overview/Getting-Started.md#running-one-app)). Do you think it's worth mentioning again?

Reply:
Aha, thank you that looks good :-)